src/gateway/handlers/channelUpdate.ts

```diff
@@ -1,28 +1,21 @@
-import { Channel } from '../../structures/channel.ts'
-import { Guild } from '../../structures/guild.ts'
-import { ChannelPayload, GuildChannelPayload } from '../../types/channel.ts'
-import getChannelByType from '../../utils/getChannelByType.ts'
-import { Gateway, GatewayEventHandler } from '../index.ts'
-
-export const channelUpdate: GatewayEventHandler = async (
-  gateway: Gateway,
-  d: ChannelPayload
-) => {
-  const oldChannel: Channel | undefined = await gateway.client.channels.get(d.id)
-
-  if (oldChannel !== undefined) {
-    await gateway.client.channels.set(d.id, d)
-    let guild: undefined | Guild;
-    if ('guild_id' in d) {
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-      guild = await gateway.client.guilds.get((d as GuildChannelPayload).guild_id) as Guild | undefined
-    }
-    if (oldChannel.type !== d.type) {
-      const channel: Channel = getChannelByType(gateway.client, d, guild) ?? oldChannel
-      gateway.client.emit('channelUpdate', oldChannel, channel)
-    } else {
-      const before = oldChannel.refreshFromData(d)
-      gateway.client.emit('channelUpdate', before, oldChannel)
-    }
-  }
-}
+import { Channel } from '../../structures/channel.ts'
+import { ChannelPayload } from "../../types/channel.ts"
+import { Gateway, GatewayEventHandler } from '../index.ts'
+
+export const channelUpdate: GatewayEventHandler = async (
+  gateway: Gateway,
+  d: ChannelPayload
+) => {
+  const oldChannel: Channel | undefined = await gateway.client.channels.get(d.id)
+  await gateway.client.channels.set(d.id, d)
+  const newChannel: Channel = (await gateway.client.channels.get(d.id) as unknown) as Channel
+
+  if (oldChannel !== undefined) {
+    // (DjDeveloperr): Already done by ChannelsManager. I'll recheck later
+    // if ('guild_id' in d) {
+    //   // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    //   (newChannel as GuildChannel).guild = await gateway.client.guilds.get((d as GuildChannelPayload).guild_id) as Guild
+    // }
+    gateway.client.emit('channelUpdate', oldChannel, newChannel)
+  } else gateway.client.emit('channelUpdateUncached', newChannel)

```

Comment:
You should emit channelUpdate even though oldChannel is undefined.

Reply:
I don't think, if `channelUpdate` is without oldChannel, it may cause troubles to user as the second argument can be undefined. I found it better to differentiate the event, just my opinion though. Though, `Uncached` suffixed events are normally rare but I added in case it goes missing.

Reply:
Okay, that makes sense.